sechub-api-java/src/main/java/com/mercedesbenz/sechub/api/java/demo/cli/SecHubClientConfigurationFactory.java

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.api.java.demo.cli;
+
+import java.net.URI;
+
+import com.mercedesbenz.sechub.api.java.SecHubClient;
+import com.mercedesbenz.sechub.api.java.demo.cli.exceptions.SecHubClientConfigurationRuntimeException;
+
+public class SecHubClientConfigurationFactory {
+
+    EnvironmentVariableReader environmentVariableReader;
+
+    public SecHubClientConfigurationFactory() {
+        environmentVariableReader = new EnvironmentVariableReader();
+    }
+
+    public SecHubClient create(CommandLineSettings settings) {
+        URI serverUri = null;
+        if (settings.getServerUri() != null)
+            serverUri = URI.create(settings.getServerUri());
+
+        int serverPort = settings.getServerPort();
+        String userId = settings.getUserId();
+        String apiToken = settings.getApiToken();
+
+        String envServerUri = environmentVariableReader.readAsString(EnvironmentVariableConstants.SECHUB_SERVER_URI);
+        if (serverUri == null && envServerUri != null) {
+            serverUri = URI.create(envServerUri);
+        }
+        if (serverPort <= 0 || serverPort > 65535) {
+            serverPort = environmentVariableReader.readAsInt(EnvironmentVariableConstants.SECHUB_SERVER_PORT);
+        }
+        if (userId == null) {
+            userId = environmentVariableReader.readAsString(EnvironmentVariableConstants.SECHUB_USERID);
+        }
+        if (apiToken == null) {
+            apiToken = environmentVariableReader.readAsString(EnvironmentVariableConstants.SECHUB_APITOKEN);
+        }
+
+        if (!isServerUriValid(serverUri)) {
+            throw new SecHubClientConfigurationRuntimeException(
+                    "Sechub server's URI is invalid. Please set the Sechub server's URI as an environment variable (SECHUB_SERVER_URI) or as a program argument (--serverUri).");
+        }
+
+        if (serverPort <= 0 || serverPort > 65535) {
+            throw new SecHubClientConfigurationRuntimeException("Sechub server's port is set to " + serverPort
+                    + ". Please set the Sechub server's port as an environment variable (SECHUB_SERVER_PORT) or as a program argument (--serverPort).");
+        }
+
+        if (userId == null) {
+            throw new SecHubClientConfigurationRuntimeException(
+                    "Sechub server's privileged user id is null. Please set the Sechub server's privileged user id as an environment variable (SECHUB_USERID) or as a program argument (--userId).");
+        }
+
+        if (apiToken == null) {
+            throw new SecHubClientConfigurationRuntimeException(
+                    "Privileged user's API token is null. Please set the privileged user's API token as an environment variable (SECHUB_APITOKEN) or as a program argument (--apiToken).");
+        }
+
+        return SecHubClient.create(userId, apiToken, serverUri.toString(), serverPort);
```

Comment:
Please change the `SecHubClient.java` create methods. They should work using an URI rather than two separate values `hostUri` and `hostPort`. This is a design flaw, it does not make a lot of sense to work with a string and integer. Java is object oriented and has the URI object as part of the standard API. 

You should be able to create the `SecHubClient` with this call:

~~~
return SecHubClient.create(userId, apiToken, serverUri);
~~~

or with

~~~
return SecHubClient.create(userId, apiToken, serverUri, trustAllCertificates);
~~~

That should feel more "natural", passing a URL.